GrpcClient/EmployeeGrpcRepository.cs

```diff
@@ -0,0 +1,286 @@
+    public async Task<Entities.Employee> GetSingleAsync(long id)
+    {
+        Entities.Employee employee = new Entities.Employee();
+        try
+        {
+            using var channel = GrpcChannel.ForAddress(_grpcAddress); 
+            var client = new Employee.EmployeeClient(channel);
+            var reply = await client.GetSingleEmployeeByIdAsync(new Id { Id_ = id }); 
+            employee = grpcEmployeeDtoToEntityEmployee(reply,
+                GrpcShiftDtosToEntityShiftsList(reply.AssignedShifts.Dtos.ToList()));
+        }
+        catch (RpcException e)
+        {
+            if (e.StatusCode == StatusCode.NotFound)
+            {
+                throw new ArgumentException(e.Message);
+            }
+        }
+
+        return employee;
+    }
+
+    public async Task<Entities.Employee> GetSingleEmployeeByWorkingNumberAsync(int WorkingNumber)
+    {
+        Entities.Employee employee = new Entities.Employee();
+        try
+        {
+            using var channel = GrpcChannel.ForAddress(_grpcAddress); 
+            var client = new Employee.EmployeeClient(channel);
+            var reply = await client.GetSingleEmployeeByWorkingNumberAsync(new WorkingNumber { WorkingNumber_ = uint.CreateChecked(WorkingNumber)  }); 
+            employee = grpcEmployeeDtoToEntityEmployee(reply, GrpcShiftDtosToEntityShiftsList(reply.AssignedShifts.Dtos.ToList()));
+        }
+        catch (RpcException e)
+        {
+            if (e.StatusCode == StatusCode.NotFound)
+            {
+                throw new ArgumentException(e.Message);
+            }
+        }
+
+        return employee;
+    }
+
+    public async Task<bool> IsEmployeeInRepository(long Id)
+    {
+        System.Boolean b = new System.Boolean();
+        
+        using var channel = GrpcChannel.ForAddress(_grpcAddress); 
+        var client = new Employee.EmployeeClient(channel);
+        var reply = await client.IsEmployeeInRepositoryAsync(new Id { Id_ = Id });
+        b = reply.Boolean_;
+        
+        return b;
+    }
+
+    public static List<Entities.Shift> GrpcShiftDtosToEntityShiftsList(List<ShiftDTO> grpcShiftDtos)
+    {
+        List<Entities.Shift> shifts = new List<Entities.Shift>();
+        foreach (var shiftDTO in grpcShiftDtos)
+        {
+            var shift = new Entities.Shift
+            {
+                Id = shiftDTO.Id,
+                StartDateTime = new DateTime(shiftDTO.StartDateTime),
+                EndDateTime = new DateTime(shiftDTO.EndDateTime),
+                TypeOfShift = shiftDTO.TypeOfShift,
+                ShiftStatus = shiftDTO.ShiftStatus,
+                Description = shiftDTO.Description,
+                Location = shiftDTO.Location,
+                AssingnedEmployees = shiftDTO.AssignedEmployeeIds.ToList()
+            };
+            shifts.Add(shift);
+        }
+
+        return shifts;
+    }
+    
+    public static List<Entities.Shift> EntityShiftDtosToEntityShiftsList(List<DTOs.Shift.ShiftDTO> shiftDtos)
```

Comment:
Do we have 2 sets of dtos? Why there are grpc dtos (previous method) and entity dtos?

Reply:
grpc dtos are coming from .proto file and they are for java and entity ones are from this csharp project